protobuf/concept.proto

```diff
@@ -32,76 +32,105 @@ message Concept {
+        message Req {
+            string label = 1;
         }
+        message Res {}
     }
-    message Res {
-        oneof res {
-            // Thing method responses
-            Thing.Delete.Res thing_delete_res = 900;
-            Thing.GetType.Res thing_getType_res = 901;
-            Thing.IsInferred.Res thing_isInferred_res = 902;
-            Thing.SetHas.Res thing_setHas_res = 906;
-            Thing.UnsetHas.Res thing_unsetHas_res = 907;
 
-            // Relation method responses
-            Relation.AddPlayer.Res relation_addPlayer_res = 1002;
-            Relation.RemovePlayer.Res relation_removePlayer_res = 1003;
+    message IsAbstract {
+        message Req {}
+        message Res {
+            bool abstract = 1;
         }
     }
 
-    message Iter {
+    message GetSupertype {
+        message Req {}
+        message Res {
+            oneof res {
+                Type type = 1;
+            }
+        }
+    }
+
+    message SetSupertype {
         message Req {
-            oneof req {
-                // Thing iterator requests
-                Thing.GetHas.Iter.Req thing_getHas_iter_req = 903;
-                Thing.GetRelations.Iter.Req thing_getRelations_iter_req = 904;
-                Thing.GetPlays.Iter.Req thing_getPlays_iter_req = 905;
+            Type type = 1;
+        }
+        message Res {}
+    }
 
-                // Relation iterator responses
-                Relation.GetPlayers.Iter.Req relation_getPlayers_iter_req = 1000;
-                Relation.GetPlayersByRoleType.Iter.Req relation_getPlayersByRoleType_iter_req = 1001;
+    message GetSupertypes {
+        message Iter {
+            message Req {}
+            message Res {
+                Type type = 1;
+            }
+        }
+    }
 
-                // Attribute iterator requests
-                Attribute.GetOwners.Iter.Req attribute_getOwners_iter_req = 1101;
+    message GetSubtypes {
+        message Iter {
+            message Req {}
+            message Res {
+                Type type = 1;
             }
         }
+    }
+}
 
+// Rule
+
+message Rule {
+    string label = 1;
+    string when = 2;
+    string then = 3;
+
+    message When {
+        message Req {}
         message Res {
             oneof res {
-                // Thing iterator responses
-                Thing.GetHas.Iter.Res thing_getHas_iter_res = 903;
-                Thing.GetRelations.Iter.Res thing_getRelations_iter_res = 904;
-                Thing.GetPlays.Iter.Res thing_getPlays_iter_res = 905;
-
-                // Relation iterator responses
-                Relation.GetPlayers.Iter.Res relation_getPlayers_iter_res = 1000;
-                Relation.GetPlayersByRoleType.Iter.Res relation_getPlayersByRoleType_iter_res = 1001;
+                string pattern = 1;
+            }
+        }
+    }
 
-                // Attribute iterator responses
-                Attribute.GetOwners.Iter.Res attribute_getOwners_iter_res = 1101;
+    message Then {
```

Comment:
Following our new method naming convention on Concept API, the methods on rules should be `getWhen()` and `getThen()`, which means the message names here needs to reflect that: `GetWhen`, and `GetThen`. Make sure to apply this onto the rest of the variable names through this proto file.

Reply:
And make sure the method names on the server and client side concept API are implemented accordingly too, @flyingsilverfin

Reply:
addressed

Reply:
The message names here are still `When` and `Then` @flyingsilverfin . Look at the other message names. They're supposed to represent the method names, right? So these ones should be `GetWhen` and `GetThen`. Also, given the discussion during our call, you should also introduce `SetWhen` and `SetThen` now, right?